app/test/e2e/alertQueue.spec.js

```diff
@@ -439,17 +439,103 @@ describe('AlertQueue ', () => {
         statistic.should.have.property('createdAt').and.be.a('date');
     });
 
+    it('Dataset messages received when resource type is EMAIL trigger an email being queued', async () => {
+        await new Subscription(createSubscription(ROLES.USER.id, 'viirs-active-fires', {
+            datasetsQuery: [{ id: 'viirs-active-fires', type: 'dataset' }],
+            resource: {
+                content: 'subscription-recipient@vizzuality.com',
+                type: 'EMAIL'
+            },
+        })).save();
 
-    afterEach(() => {
+        nock(process.env.CT_URL)
+            .get('/v1/dataset/viirs-active-fires')
+            .reply(200, {
+                data: {
+                    attributes: {
+                        subscribable: { dataset: { subscriptionQuery: '' } },
+                        tableName: 'test',
+                    }
+                }
+            });
+
+        nock(process.env.CT_URL)
+            .get('/v1/query')
+            .query(() => true)
+            .reply(200, { data: [{ value: 10000 }] });
+
+        nock(process.env.CT_URL)
+            .get('/v1/dataset/viirs-active-fires/metadata')
+            .query(() => true)
+            .reply(200, { data: [{ attributes: { info: { name: 'metatest' } } }] });
+
+        process.on('unhandledRejection', (error) => { should.fail(error); });
+
+        redisClient.on('message', (channel, message) => {
+            const jsonMessage = JSON.parse(message);
+            jsonMessage.should.have.property('template');
+            if (jsonMessage.template === 'dataset-rw') {
+                jsonMessage.should.have.property('sender').and.equal('rw');
+                jsonMessage.should.have.property('data').and.be.a('object');
+                jsonMessage.should.have.property('recipients').and.be.a('array').and.have.length(1);
+                jsonMessage.recipients[0].should.have.property('address').and.equal('subscription-recipient@vizzuality.com');
+            } else {
+                should.fail('Unsupported message type: ', jsonMessage.template);
+            }
+        });
+
+        await AlertQueue.processMessage(null, JSON.stringify({ layer_slug: 'dataset' }));
+    });
+
+    it('Dataset messages received when resource type is URl trigger a POST request to a web-hook URL', async () => {
```

Comment:
add tests for more than the happy case

Reply:
Added tests for cases such as invalid URL and POSTing to a web-hook that returns 4XX codes. Although at the moment this is not very useful (test-wise), since we are only logging messages in those cases - so no testable output.

I tried to ensure log messages were added in these test cases, but I wasn't able to fetch messages from our logging lib.

Reply:
I'm ok with the tests not validating the log messages. However, do make sure that these scenarios are well logged. If someone comes to us and says "hey, my webhooks are not being called, can you check it please?" we should have the necessary logs to quickly identify if the webhooks are called or not, and what's the return of the call.

Reply:
The `subscription` object (recently added) and the `body` of the response to the POST request, along with the current logs, should enable to easily identify who made the request and the return of the webhook call :+1: